### PR TITLE
feat: add rescue lifecycle and primary recovery in Doctor

### PR DESCRIPTION
**Pull request by Keith-CY**
## Summary
- add full Rescue Bot lifecycle management in Tauri (local + SSH): `set | activate | status | deactivate | unset`
- make rescue operations more robust and idempotent:
  - activation restart-timeout fallback (`restart` -> `stop/start`)
  - cleanup no-op tolerance for deactivate/unset (already stopped/uninstalled/unset)
- add Rescue-driven primary health flow:
  - diagnosis result with checks/issues panel
  - safe auto-repair pipeline with re-check
- remove manual dependency for primary doctor JSON mode:
  - auto fallback from `openclaw --profile <profile> doctor --json` to `openclaw --profile <profile> doctor` when `--json` is unsupported
- improve Doctor UI interactions:
  - status-first Rescue card with `Activate`, `Deactivate`, `Unset`
  - quick-fix actions directly on failed checks and auto-fixable issue items
  - clearer localized messages (EN/ZH)

## Test Plan
- [x] `cargo check --quiet` (in `src-tauri/`)
- [x] `bun run tsc --noEmit`
- [ ] `cargo test rescue_bot_tests --quiet` blocked in this environment by DNS resolution to `static.crates.io`

## Notes
- Doctor remains the primary entrypoint for Rescue lifecycle operations.
- Auto-repair still only applies to explicitly mapped safe primary issues.

### CODE REVIEW

**Review by dev01lay2 — COMMENTED** [review]
## Review Summary

Solid feature addition — Rescue Bot lifecycle + primary recovery diagnostics are well-structured. The local/remote symmetry is clean. A few things to address:

### Issues

**1. `run_openclaw_dynamic` duplicates `OpenclawCommandOutput` fields unnecessarily**
`commands.rs` — `run_openclaw_dynamic` takes the output from `run_openclaw` (which already returns `OpenclawCommandOutput`) and reconstructs it field-by-field. If the types match, just return it directly.

**2. `json.syntax` fix is a no-op pretending to be a fix**
`build_primary_issue_fix_command` maps `json.syntax` to the same command as `field.agents` (set `agents.defaults.model`). The original `remote_fix_issues` just marks it applied without doing anything meaningful either. If the intent is "normalize the config by rewriting it," consider a dedicated `config normalize` command or at least add a comment explaining why this works.

**3. `collect_model_summary` moved but unchanged — noisy diff**
The function and several hundred lines of helpers (`RescueBotAction`, utility fns) were inserted in the middle of existing code, causing a large delete+reinsert of `fix_issues` / `collect_model_summary`. The actual logic is unchanged — consider reordering to minimize churn (new code at end of file, before tests).

**4. Missing `use_effect` cleanup / abort for async operations in Doctor.tsx**
`refreshRescueStatus` fires on mount via `useEffect` but has no abort/cleanup. If the component unmounts mid-flight (instance switch), the setState calls will fire on an unmounted component. Consider an abort flag or `AbortController`.

**5. State explosion in Doctor.tsx**
15+ individual `useState` hooks for rescue/primary state. Consider grouping into a reducer or at least an object state to make resets cleaner and reduce the reset list in the instance-change effect.

### Minor / Nits

- `let _ = new_text;` in `remote_fix_issues` is dead code (was there before, but worth cleaning up since you're touching the file)
- `rescue_port` in `RescuePrimaryDiagnosisResult` is `Option<u16>` on Rust side but `rescuePort?: number` in TS — fine, but document the mapping for optional fields vs undefined
- Test `test_suggest_rescue_port_prefers_large_gap` only tests one case — consider edge cases (port near u16::MAX, port 0)
- `cargo test rescue_bot_tests` blocked by DNS — noted in PR description, but ensure CI has the fix before merge

File: src-tauri/src/commands.rs [hunk 1]
```diff
@@ -1945,104 +2140,1238 @@ pub fn fix_issues(ids: Vec<String>) -> Result<FixResult, String> {
...
+            .get("autoFixable")
+            .and_then(Value::as_bool)
+            .or_else(|| obj.get("auto_fixable").and_then(Value::as_bool))
+            .unwrap_or(false);
+        let fix_hint = obj
+            .get("fixHint")
+            .and_then(Value::as_str)
+            .or_else(|| obj.get("fix_hint").and_then(Value::as_str))
+            .map(str::to_string);
+        items.push(RescuePrimaryIssue {
```
Comment (dev01lay2) [inline src-tauri/src/commands.rs:2380]: `run_openclaw_dynamic` reconstructs `OpenclawCommandOutput` field-by-field from the return of `run_openclaw`, which already returns the same struct. If the types truly match, just return it directly instead of destructuring and rebuilding.

File: src-tauri/src/commands.rs [hunk 2]
```diff
@@ -1945,104 +2140,1238 @@ pub fn fix_issues(ids: Vec<String>) -> Result<FixResult, String> {
...
+}
+
+fn command_detail(output: &OpenclawCommandOutput) -> String {
+    if !output.stderr.trim().is_empty() {
+        return trim_for_detail(&output.stderr);
+    }
+    if !output.stdout.trim().is_empty() {
+        return trim_for_detail(&output.stdout);
+    }
+    "no output".into()
```
Comment (dev01lay2) [inline src-tauri/src/commands.rs:2318]: This maps `json.syntax` to the exact same command as `field.agents` (`config set agents.defaults.model`). If the intent is "rewrite the config file to normalize JSON syntax," that's a side-effect of `config set` — worth a comment explaining why, otherwise it looks like a copy-paste error.

File: src/pages/Doctor.tsx [hunk 3]
```diff
@@ -50,11 +55,41 @@ export function Doctor({ sshHosts }: DoctorProps) {
   const [logsContent, setLogsContent] = useState("");
   const [logsLoading, setLogsLoading] = useState(false);
   const logsContentRef = useRef<HTMLPreElement>(null);
+  const [rescueActivating, setRescueActivating] = useState(false);
```
Comment (dev01lay2) [inline src/pages/Doctor.tsx]: 15+ individual `useState` hooks for rescue/primary state makes the reset in the instance-change effect fragile (easy to miss one). Consider `useReducer` or grouping into `useState<RescueState>(initialState)` so reset is just `setRescueState(initialState)`.

File: src/pages/Doctor.tsx [hunk 4]
```diff
@@ -165,6 +200,249 @@ export function Doctor({ sshHosts }: DoctorProps) {
...
+    } finally {
+      setPrimaryRepairing(false);
+      setPrimaryRepairingIssueId(null);
+    }
+  };
+
+  useEffect(() => {
+    void refreshRescueStatus();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [instanceId, isRemote, isConnected]);
```
Comment (dev01lay2) [inline src/pages/Doctor.tsx:547]: This `useEffect` fires an async `refreshRescueStatus()` but has no cleanup. If `instanceId` changes quickly (user switching instances), the old call's `setState` will land on stale state. Add a mounted/aborted flag:
```ts
useEffect(() => {
  let cancelled = false;
  refreshRescueStatus(cancelled);
  return () => { cancelled = true; };
}, [instanceId, isRemote, isConnected]);
```

File: src-tauri/src/commands.rs [hunk 5]
```diff
@@ -1945,104 +2150,1228 @@ pub fn fix_issues(ids: Vec<String>) -> Result<FixResult, String> {
...
+}
+
+fn parse_json_loose(raw: &str) -> Option<Value> {
+    if raw.trim().is_empty() {
+        return None;
+    }
+    serde_json::from_str(raw)
+        .ok()
+        .or_else(|| extract_json_from_output(raw).and_then(|json| serde_json::from_str(json).ok()))
```
[the remark this answers is not in the document]
Comment (Keith-CY) [reply src-tauri/src/commands.rs:2327]: Good catch — patched in bece945.\n\nWhat changed:\n- `extract_json_from_output` now extracts the trailing balanced JSON payload by scanning backward from the last `}`/`]` and matching its opener, instead of slicing from the first `{`/`[`.\n- This handles prefixed bracketed log lines (for example `[plugins] ...`) before the real JSON payload.\n- Added tests for both helpers:\n  - `test_extract_json_from_output_uses_trailing_balanced_payload`\n  - `test_parse_json_loose_handles_leading_bracketed_logs`\n\nRe-validated with `cargo check --quiet` and `bun run tsc --noEmit`.
